### PR TITLE
New Extension: Raynab

### PR DESCRIPTION
**Pull request by gogocharli**
## Description

With the help of the YNAB API, Raynab allows users of Raycast to manage their budgets completely from inside the application. The main commands of the extensions are:

- Select Active Budget
- View Active Budget
- List Transactions
- Create Transaction
- List Accounts

## Screencast

https://cln.sh/zpT95y

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/raynab/package.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  "name": "raynab",
+  "title": "Raynab — Manage Your Budgets",
+  "description": "View and manage your YNAB Budgets directly from Raycast",
+  "icon": "command-icon.png",
+  "author": "CodetaroMiura",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "activeBudget",
+      "title": "Select Active Budget",
```
Comment (pernielsentikaer) [inline extensions/raynab/package.json:12]:
```suggestion
      "title": "Select Active Budget",
      "subtitle": "YNAB",
```

File: extensions/raynab/package.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      "description": "Select an active budget to use across Raynab.",
+      "mode": "view"
+    },
+    {
+      "name": "transactions",
+      "title": "List Transactions",
```
Comment (pernielsentikaer) [inline extensions/raynab/package.json:18]:
```suggestion
      "title": "List Transactions",
      "subtitle": "YNAB",
```

File: extensions/raynab/package.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      "description": "List transactions for the selected budget.",
+      "mode": "view"
+    },
+    {
+      "name": "transaction",
+      "title": "Create Transaction",
```
Comment (pernielsentikaer) [inline extensions/raynab/package.json:24]:
```suggestion
      "title": "Create Transaction",
      "subtitle": "YNAB",
```

File: extensions/raynab/package.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      "description": "Create a new transaction in the selected budget.",
+      "mode": "view"
+    },
+    {
+      "name": "accounts",
+      "title": "List Accounts",
```
Comment (pernielsentikaer) [inline extensions/raynab/package.json:30]:
```suggestion
      "title": "List Accounts",
      "subtitle": "YNAB",
```

File: extensions/raynab/package.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      "description": "List accounts for the selected budget.",
+      "mode": "view"
+    },
+    {
+      "name": "budget",
+      "title": "View Active Budget",
```
Comment (pernielsentikaer) [inline extensions/raynab/package.json:36]:
```suggestion
      "title": "View Active Budget",
      "subtitle": "YNAB",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Wow, @gogocharli! 🚀 

What an extension, it's hard to test it fully – also, never tried YNAB before but everything I've tested is working 😄 

I gave it a good long spin but couldn't really find anything in the first run 🔥 

- One thing I noticed is that all amounts in the extension is in USD, I think we should remove that – or if possible (maybe too hard?), get the currency from the app and show (then it's not localization) 

What do you think? 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you want to discuss if you have any questions.

**Comment by gogocharli** [comment]
Hello @pernielsentikaer, thank you for the review!

> One thing I noticed is that all amounts in the extension is in USD, I think we should remove that – or if possible (maybe too hard?), get the currency from the app and show (then it's not localization)

This is already possible. But YNAB defaults that setting to USD by default for the budgets.

See below when I change my test account to Euros.

CleanShot 2022-03-04 at 16 22 22
CleanShot 2022-03-04 at 16 23 14
CleanShot 2022-03-04 at 16 23 23

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @gogocharli 👋 

You're on 🔥 What an extension! For me, it's such a powerful showcase you're doing here!

I think I've missed the that part, after changing the currency I had to “Set Active Budget” again in order to get the right values within the extension 😄 

Is there a way to add the currency to `List Transactions`, seems like it's one of the few places where it's not present?

I'm not sure if it's possible, but it could be cool if `Currency Placement` from YNAB was used within the extension, but it's a NIT

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by gogocharli** [comment]
Hi again @pernielsentikaer, thank you for the suggestion!

I had previously decided not to show the transactions list because it seemed too distracting in the list view. But, you're right, it is more consistent with the app.

I have used the `CurrencyFormat` setting within YNAB to make the changes as suggested 😄 

**Symbol after amount**

CleanShot 2022-03-05 at 09 49 57@2x

**Symbol before amount**

CleanShot 2022-03-05 at 09 48 42@2x

**Symbol before amount (negative)**

CleanShot 2022-03-05 at 10 04 53@2x

**No Symbol**

CleanShot 2022-03-05 at 09 48 01@2x

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Another great update @gogocharli 🔥 

We're close to merging, I have the last feedback before we're ready to ship it 💪

- When selecting an active budget, it would then be nice with a Toast showing a status, right now it feels like it doesn't work 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by gogocharli** [comment]
Thank you for another great suggestion, @pernielsentikaer.

This is now updated!

**Review by pernielsentikaer — APPROVED** [review]
Hi @gogocharli 👋 

This extension is another great example of what's possible to archive with the API – well done! I'm out of ideas, but still a bit blown about the extension 😄 

Approved, it has been a pleasure working with you 🔥

**Pull request merged**